widget/textgrid.go

```diff
@@ -84,34 +86,48 @@ func (t *TextGrid) Resize(size fyne.Size) {
 // SetText updates the buffer of this textgrid to contain the specified text.
 // New lines and columns will be added as required. Lines are separated by '\n'.
 // The grid will use default text style and any previous content and style will be removed.
+// Tab characters are padded with spaces to the next tab stop.
 func (t *TextGrid) SetText(text string) {
 	lines := strings.Split(text, "\n")
 	rows := make([]TextGridRow, len(lines))
 	for i, line := range lines {
-		spaced := strings.ReplaceAll(line, "\t", textTabIndent)
-
-		cells := make([]TextGridCell, len(spaced))
-		extras := 0
-		for j, r := range line {
-			cells[j+extras] = TextGridCell{Rune: r}
-
+		cells := make([]TextGridCell, 0, len(line))
+		for _, r := range line {
+			cells = append(cells, TextGridCell{Rune: r})
 			if r == '\t' {
-				for k := j + extras + 1; k < j+extras+len(textTabIndent); k++ {
-					cells[k] = TextGridCell{Rune: ' '}
+				col := len(cells)
+				next := nextTab(col-1, t.TabWidth())
+				for i := col; i < next; i++ {
+					cells = append(cells, TextGridCell{Rune: ' '})
 				}
-				extras += len(textTabIndent) - 1
 			}
 		}
-
 		rows[i] = TextGridRow{Cells: cells}
 	}
 
 	t.Rows = rows
 	t.Refresh()
 }
 
+// TabWidth either returns the set tab width or if not set the returns the DefaultTabWidth
+func (t *TextGrid) TabWidth() int {
```

Comment:
As above, we could just use the field instead of two new methods.

Reply:
done